pkg/logql/log/filter.go

```diff
@@ -425,6 +439,10 @@ func simplify(reg *syntax.Regexp) (Filterer, bool) {
 		if reg.Sub[0].Op == syntax.OpAnyCharNotNL {
 			return TrueFilter, true
 		}
+	case syntax.OpPlus:
+		if len(reg.Sub) == 1 && reg.Sub[0].Op == syntax.OpAnyCharNotNL { // simplify ".+"
```

Comment:
```suggestion
		if len(reg.Sub) >= 1 && reg.Sub[0].Op == syntax.OpAnyCharNotNL { // simplify ".+"
```

no idea if the suggestion makes sense but maybe we can have multiple subexpressions and the simplification still be valid

Reply:
There's definitely room for improvement but it gets more difficult to understand what the expression is doing when there are more sub expressions of `+` than just a single `.`